lib/rspec/core/reporter.rb

```diff
@@ -152,6 +152,18 @@ def deprecation(hash)
     end
 
     # @private
+    # Provides a way to notify of an exception that is not tied to any
+    # particular exception (such as an exception encountered in a :suite hook).
+    # Exceptions will be formatted the same way they normally are.
+    def notify_non_example_exception(exception, context_description)
+      @configuration.world.non_example_failure = true
+
+      example = Example.new(AnonymousExampleGroup, context_description, {})
+      presenter = Formatters::ExceptionPresenter.new(exception, example, :indentation => 0)
+      message presenter.fully_formatted(nil)
```

Comment:
just double checking that `presenter` here isn't an extension point for plugins or anything like that? Passing `nil` is ok?